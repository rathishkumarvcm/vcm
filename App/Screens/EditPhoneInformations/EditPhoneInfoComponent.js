import React, { Component } from 'react';
import { Text, View, Image, ScrollView, TouchableOpacity, FlatList } from 'react-native';
import { styles } from './styles';
import { GButtonComponent, GHeaderComponent, GIcon, GInputComponent, GRadioButtonComponent } from '../../CommonComponents';
import { scaledHeight } from '../../Utils/Resolution';
import globalString from '../../Constants/GlobalStrings';
import PropTypes from "prop-types";

const tempUserMobile = [
    {
        "mobileNumberType": 'Primary Mobile',
        "mobileNumber": '+1(xxx) xxx-7890',
        "mobilePreferredTime": 'Morning',
        "isPrimaryMobile": true
    },
    {
        "mobileNumberType": 'Secondary Mobile',
        "mobileNumber": '+1(xxx) xxx-7890',
        "mobilePreferredTime": 'Morning',
        "isPrimaryMobile": false
    },
    {
        "mobileNumberType": 'Teritary Mobile',
        "mobileNumber": '+1(xxx) xxx-7890',
        "mobilePreferredTime": 'Morning',
        "isPrimaryMobile": false
    }
];

const tempUserHome = [
    {
        "mobileNumberType": 'Primary Mobile',
        "mobileNumber": '+1(xxx) xxx-7890',
        "mobilePreferredTime": 'Morning',
        "isPrimaryMobile": true
    },
    {
        "mobileNumberType": 'Secondary Mobile',
        "mobileNumber": '+1(xxx) xxx-7890',
        "mobilePreferredTime": 'Morning',
        "isPrimaryMobile": false
    }
];

const tempUserWork = [
    {
        "mobileNumberType": 'Primary Mobile',
        "mobileNumber": '+1(xxx) xxx-7890',
        "mobilePreferredTime": 'Morning',
        "isPrimaryMobile": true
    }
];

const UserPhoneInformation = (props) => {
    return (
        <View style={styles.editEmailHolder}>
            <Text style={styles.editEmailType}>
                {props.mobileNumberType}
            </Text>

            <Text style={styles.editEmailId}>
                {props.mobileNumber}
            </Text>

            <Text style={styles.editEmailId}>
                {props.mobilePreferredTime}
            </Text>

            <View style={styles.editEmailBorder} />

            <View style={styles.editEmailPrimaryContent}>
                <Text style={styles.editEmailId}>
                    {globalString.profileSettingsPage.profileMailPrimaryLabel}
                </Text>
            </View>
        </View>
    );
};

UserPhoneInformation.propTypes = {
    mobileNumberType: PropTypes.string,
    mobileNumber: PropTypes.string,
    mobilePreferredTime: PropTypes.string
};

class editPhoneInfoComponent extends Component {
    constructor(props) {
        super(props);
        //set true to isLoading if data for this screen yet to be received and wanted to show loader.
        this.state = {
            isLoading: false,
            enableBiometric: false,
            faceIdEnrolled: false,
            touchIdEnrolled: false
        };
    }

    renderPhoneInformation = (dataLength) => ({ item, index }) =>
        (<UserPhoneInformation
            mobileNumberType={item.mobileNumberType}
            mobileNumber={item.mobileNumber}
            mobilePreferredTime={item.mobilePreferredTime}
        />);

    componentDidMount() { }

    phoneInformationOnAdd = () => this.props.navigation.navigate('editAddPhoneNumber');

    phoneInformationOnCancel = () => this.props.navigation.navigate('profileSettings');

    render() {
        return (
            <View style={styles.container}>
                <GHeaderComponent
                    navigation={this.props.navigation}
                />

                <ScrollView style={{ flex: 0.85 }}>

                    <View style={styles.settingsView}>
                        <Text style={styles.settingsInfo}>
                            {globalString.editProfilePageValue.editAddressInfoHead}
                        </Text>
                        <Text style={[styles.settingsInfo, styles.editLabelBold]}>
                            {globalString.editPhoneInformations.phoneInfoTitle}
                        </Text>
                    </View>

                    <View style={styles.settingsView}>
                        <Text style={styles.phoneInfoLabel}>
                            {globalString.editPhoneInformations.phoneLabel}
                        </Text>
                        <Text style={styles.phoneInfoAddNewLabel}
                            onPress={this.phoneInformationOnAdd}
                        >
                            {globalString.editPhoneInformations.phoneAddNew}
                        </Text>
                    </View>

                    <View style={styles.settingsBorder} />

                    <View>
                        <View style={styles.settingsView}>
                            <Text style={styles.phoneMobileView}>
                                {globalString.editPhoneInformations.phoneMobileLabel}
                            </Text>
                        </View>

                        <FlatList
                            data={tempUserMobile}
                            keyExtractor={this.generateKeyExtractor}
                            renderItem={this.renderPhoneInformation(tempUserMobile.length)}
                        />

                    </View>

                    <View>
                        <View style={styles.settingsView}>
                            <Text style={styles.phoneMobileView}>{"Home"}</Text>
                        </View>

                        <FlatList
                            data={tempUserHome}
                            keyExtractor={this.generateKeyExtractor}
                            renderItem={this.renderPhoneInformation(tempUserHome.length)}
                        />

                    </View>

                    <View>
                        <View style={styles.settingsView}>
                            <Text style={styles.phoneMobileView}>{"Work"}</Text>
                        </View>

                        <FlatList
                            data={tempUserWork}
                            keyExtractor={this.generateKeyExtractor}
                            renderItem={this.renderPhoneInformation(tempUserWork.length)}
                        />

                    </View>

                    <View style={styles.settingsView}>
                        <Text style={styles.phoneAddFaxLabel}>
                            {globalString.editPhoneInformations.phoneFax}
                        </Text>
                        <Text style={styles.phoneFaxAddNewLabel}>
                            {globalString.editPhoneInformations.phoneAddNew}
                        </Text>
                    </View>

                    <View style={styles.settingsBorder} />

                    <View style={styles.phoneFaxView}>
                        <GInputComponent
                            placeholder={globalString.editPhoneInformations.phoneFaxLabel}
                        />
                    </View>

                    <View style={styles.editFlexDirectionColumn}>
                        <GButtonComponent
                            buttonStyle={styles.cancelButtonStyle}
                            buttonText={globalString.common.cancel}
                            textStyle={styles.cancelButtonText}
                            onPress={this.phoneInformationOnCancel}
                        />
                    </View>

                    <View style={styles.newVictorySection}>
                        <Text style={styles.termsofuseText1}>
                            {globalString.editPhoneInformations.phoneTerms}
                        </Text>
                        <Text style={styles.openInvestment}>
                            {globalString.editPhoneInformations.phoneInvestment}
                        </Text>
                    </View>

                    <View style={styles.connectWithUs}>
                        <Image
                            source={require("../../Images/logo.png")}
                        />
                    </View>

                    <View style={styles.whiteBackground}>
                        <Text style={styles.editLabelInputMedium}>
                            {globalString.common.connectWithUs}
                        </Text>
                    </View>

                    <View style={styles.whiteBackground}>
                        <Image style={styles.imageWidthHeight}
                            source={require("../../Images/twitterlogo.png")}
                        />
                        <Image style={styles.imageWidthHeight}
                            source={require("../../Images/linkedinlogo.png")}
                        />
                    </View>

                    <View style={styles.privacyAgreement}>
                        <Text style={styles.privacyText}>
                            {globalString.common.privacyPolicy}
                        </Text>

                        <Text style={styles.privacyText}>
                            {globalString.common.fundDocuments}
                        </Text>
                    </View>

                    <View style={styles.privacyAgreement}>
                        <Text style={styles.privacyText}>
                            {globalString.common.userAgreement}
                        </Text>

                        <Text style={styles.privacyText}>
                            {globalString.common.support}
                        </Text>
                    </View>

                    <View style={styles.copyRightSection}>
                        <Text style={styles.copyRightText}>
                            {globalString.common.copyRights}
                        </Text>
                    </View>

                </ScrollView>
            </View>
        );
    }
}

export default editPhoneInfoComponent;