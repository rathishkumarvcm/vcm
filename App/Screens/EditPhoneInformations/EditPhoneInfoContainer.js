import { connect } from "react-redux";
import editPhoneInfoComponent from './EditPhoneInfoComponent';

const mapStateToProps = (state /* , props */) => (
  {
  }
);

const mapDispatchToProps = {
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(editPhoneInfoComponent);