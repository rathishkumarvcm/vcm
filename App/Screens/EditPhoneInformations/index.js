import editPhoneInfoComponent from './EditPhoneInfoContainer';

export default editPhoneInfoComponent;