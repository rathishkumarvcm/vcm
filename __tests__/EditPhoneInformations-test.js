import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import EditPhoneInformations from '../App/Screens/EditPhoneInformations/EditPhoneInfoComponent';

const navigation = { navigate: jest.fn(), getParam: jest.fn()};

describe('Edit Phone Informations', () => {

    // Application Screnshot
    it('snap shot', () => {
        const tree = renderer.create(<EditPhoneInformations navigation={navigation} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    // Application Render
    test('render',()=>{
        const component = shallow(<EditPhoneInformations navigation={navigation} />);  
        component.render();
      });
});