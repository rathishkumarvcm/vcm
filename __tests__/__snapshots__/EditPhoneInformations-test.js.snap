// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit Phone Informations snap shot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F7FAFF",
      "flex": 1,
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    clickable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "flex": 0.12,
        "flexDirection": "row",
        "opacity": 1,
        "paddingLeft": "4%",
        "paddingRight": "4%",
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": 100.5,
          "opacity": 1,
          "width": "25%",
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../App/Images/logo.png",
          }
        }
        style={
          Object {
            "alignItems": "flex-start",
            "justifyContent": "flex-start",
            "width": "100%",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "width": "12%",
        }
      }
    />
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginTop": 24.5,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "height": 40,
            "width": 56,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 40,
              },
              Object {
                "backgroundColor": "transparent",
                "marginHorizontal": 5,
              },
              Object {
                "fontFamily": "EvilIcons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="iconIcon"
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "width": "10%",
        }
      }
    />
    <View
      accessible={true}
      clickable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "height": 30,
            "width": 44,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 30,
              },
              Object {
                "backgroundColor": "transparent",
                "marginHorizontal": 5,
              },
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="iconIcon"
        >
          
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "height": 30,
            "width": 44,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 30,
              },
              Object {
                "backgroundColor": "transparent",
                "marginHorizontal": 5,
              },
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="iconIcon"
        >
          
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      clickable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "flex-end",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <RNGestureHandlerButton
        collapsable={false}
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        rippleColor={0}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            undefined,
          ]
        }
      >
        <View
          accessible={true}
          style={
            Object {
              "alignItems": "flex-end",
              "justifyContent": "center",
              "marginEnd": 10,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "height": 40,
                "width": 56,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "black",
                    "fontSize": 40,
                  },
                  Object {
                    "backgroundColor": "transparent",
                    "marginHorizontal": 5,
                  },
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="iconIcon"
            >
              
            </Text>
          </View>
        </View>
      </RNGestureHandlerButton>
    </View>
  </View>
  <RCTScrollView
    style={
      Object {
        "flex": 0.85,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginTop": 29.5,
            "paddingLeft": "4%",
            "paddingRight": "4%",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#B2B2B2",
              "fontSize": 26.5,
            }
          }
        >
          Settings &gt; Personal info &gt; 
        </Text>
        <Text
          style={
            Object {
              "color": "#B2B2B2",
              "fontSize": 26.5,
              "fontWeight": "bold",
            }
          }
        >
          Personal Profile Phone
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginTop": 29.5,
            "paddingLeft": "4%",
            "paddingRight": "4%",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#56565A",
              "fontSize": 29.5,
              "fontWeight": "bold",
              "width": "70%",
            }
          }
        >
          Manage Phone Information
        </Text>
        <Text
          onPress={[Function]}
          style={
            Object {
              "color": "#0000FF",
              "fontSize": 23,
              "textAlign": "right",
              "width": "30%",
            }
          }
        >
          Add
        </Text>
      </View>
      <View
        style={
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#B2B2B2",
            "marginLeft": "4%",
            "marginRight": "4%",
            "marginTop": 16.5,
          }
        }
      />
      <View>
        <View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "marginTop": 29.5,
                "paddingLeft": "4%",
                "paddingRight": "4%",
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#333333DE",
                  "fontSize": 29.5,
                  "fontWeight": "bold",
                }
              }
            >
              Mobile
            </Text>
          </View>
          <View
            style={
              Object {
                "backgroundColor": "#fff",
                "borderColor": "#CCCCCC",
                "borderWidth": 1,
                "marginBottom": "2%",
                "marginLeft": "2%",
                "marginRight": "2%",
                "marginTop": "2%",
                "padding": "4%",
                "width": "96%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#56565A",
                  "fontSize": 26.5,
                  "fontWeight": "bold",
                  "marginBottom": "2%",
                  "marginLeft": "4%",
                  "marginRight": "4%",
                  "marginTop": "4%",
                  "width": "92%",
                }
              }
            >
              None on file
            </Text>
            <Text
              style={
                Object {
                  "color": "#56565A",
                  "fontSize": 26.5,
                  "marginBottom": "2%",
                  "marginLeft": "4%",
                  "marginRight": "4%",
                  "marginTop": "4%",
                  "width": "92%",
                }
              }
            >
              Please go to basic information page to Add home number
            </Text>
          </View>
        </View>
        <View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "marginTop": 29.5,
                "paddingLeft": "4%",
                "paddingRight": "4%",
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#333333DE",
                  "fontSize": 29.5,
                  "fontWeight": "bold",
                }
              }
            >
              Home
            </Text>
          </View>
          <View
            style={
              Object {
                "backgroundColor": "#fff",
                "borderColor": "#CCCCCC",
                "borderWidth": 1,
                "marginBottom": "2%",
                "marginLeft": "2%",
                "marginRight": "2%",
                "marginTop": "2%",
                "padding": "4%",
                "width": "96%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#56565A",
                  "fontSize": 26.5,
                  "fontWeight": "bold",
                  "marginBottom": "2%",
                  "marginLeft": "4%",
                  "marginRight": "4%",
                  "marginTop": "4%",
                  "width": "92%",
                }
              }
            >
              None on file
            </Text>
            <Text
              style={
                Object {
                  "color": "#56565A",
                  "fontSize": 26.5,
                  "marginBottom": "2%",
                  "marginLeft": "4%",
                  "marginRight": "4%",
                  "marginTop": "4%",
                  "width": "92%",
                }
              }
            >
              Please go to basic information page to Add home number
            </Text>
          </View>
        </View>
        <View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "marginTop": 29.5,
                "paddingLeft": "4%",
                "paddingRight": "4%",
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#333333DE",
                  "fontSize": 29.5,
                  "fontWeight": "bold",
                }
              }
            >
              Work
            </Text>
          </View>
          <View
            style={
              Object {
                "backgroundColor": "#fff",
                "borderColor": "#CCCCCC",
                "borderWidth": 1,
                "marginBottom": "2%",
                "marginLeft": "2%",
                "marginRight": "2%",
                "marginTop": "2%",
                "padding": "4%",
                "width": "96%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#56565A",
                  "fontSize": 26.5,
                  "fontWeight": "bold",
                  "marginBottom": "2%",
                  "marginLeft": "4%",
                  "marginRight": "4%",
                  "marginTop": "4%",
                  "width": "92%",
                }
              }
            >
              None on file
            </Text>
            <Text
              style={
                Object {
                  "color": "#56565A",
                  "fontSize": 26.5,
                  "marginBottom": "2%",
                  "marginLeft": "4%",
                  "marginRight": "4%",
                  "marginTop": "4%",
                  "width": "92%",
                }
              }
            >
              Please go to basic information page to Add home number
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "marginTop": 29.5,
              "paddingLeft": "4%",
              "paddingRight": "4%",
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#707070",
                "fontSize": 29.5,
                "fontWeight": "bold",
                "marginRight": "4%",
                "width": "70%",
              }
            }
          >
            Fax
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "column",
              "marginBottom": "2%",
              "marginLeft": "4%",
              "marginRight": "4%",
              "marginTop": "3%",
              "width": "100%",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#FFFFFF",
                  "borderColor": "#DEDEDF",
                  "borderRadius": 6.5,
                  "borderWidth": 1,
                  "height": 79,
                  "justifyContent": "center",
                  "width": "92%",
                },
                undefined,
              ]
            }
          >
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              autoCorrect={false}
              autoFocus={false}
              editable={true}
              inputStyle={Object {}}
              keyboardType="default"
              maxLength={100}
              multiline={false}
              numberOfLines={1}
              onBlur={null}
              onChange={null}
              onChangeText={null}
              onSubmitEditing={null}
              placeholder="Fax Number"
              placeholderTextColor={null}
              rejectResponderTermination={true}
              returnKeyType="next"
              selectionColor={null}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "color": "#B5B5B5",
                    "fontSize": 26.5,
                    "justifyContent": "center",
                    "marginLeft": 23,
                  },
                  Object {},
                ]
              }
              underlineColorAndroid="transparent"
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "column",
            "marginTop": 33,
            "paddingLeft": "4%",
            "paddingRight": "4%",
            "width": "100%",
          }
        }
      >
        <View
          accessible={true}
          clickable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderColor": "#56565A",
              "borderRadius": 1,
              "borderWidth": 1,
              "height": 82,
              "justifyContent": "center",
              "marginLeft": "4%",
              "marginRight": "4%",
              "marginTop": 19.5,
              "opacity": 1,
              "width": "92%",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#56565A",
                  "fontSize": 26.5,
                },
              ]
            }
          >
            Cancel
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "marginTop": 39.5,
            "padding": 24.5,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#56565A",
              "fontSize": 29.5,
              "marginBottom": 8,
            }
          }
        >
          Investing involves risk including loss of principal.
        </Text>
        <Text
          style={
            Object {
              "color": "#56565A",
              "fontSize": 26.5,
              "justifyContent": "center",
            }
          }
        >
          Victory Mutual Funds and USAA Mutual Funds are distributed by Victory Capital Advisers, Inc. (VCA). VictoryShares ETFs and VictoryShares USAA ETFs are distributed by Foreside Fund Services, LLC (Foreside). VCA and Foreside are members of FINRA and SIPC. Victory Capital Management Inc. (VCM) is the investment adviser to the Victory Mutual Funds, USAA Mutual Funds, VictoryShares ETFs and VictoryShares USAA ETFs. VCA and VCM are not affiliated with Foreside. USAA is not affiliated with Foreside, VCM, or VCA. USAA and the USAA logos are registered trademarks and the USAA Mutual Funds and USAA Investments logos are trademarks of United Services Automobile Association and are being used by Victory Capital and its affiliates under license. Victory Capital means Victory Capital Management Inc., the investment manager of the USAA 529 College Savings Plan (Plan). The Plan is distributed by Victory Capital Advisers, Inc., a broker dealer registered with FINRA and an affiliate of Victory Capital. Victory Capital and its affiliates are not affiliated with United Services Automobile Association or its affiliates. USAA and the USAA logo are registered trademarks and the USAA 529 College Savings Plan logo is a trademark of United Services Automobile Association and are being used by Victory Capital and its affiliates under license.
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "flexDirection": "column",
            "paddingBottom": "5%",
            "paddingLeft": "5%",
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../App/Images/logo.png",
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "flexDirection": "row",
            "paddingLeft": "5%",
            "paddingRight": "5%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#56565A",
              "fontSize": 29.5,
            }
          }
        >
          Connect with Us
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "flexDirection": "row",
            "paddingLeft": "5%",
            "paddingRight": "5%",
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../App/Images/twitterlogo.png",
            }
          }
          style={
            Object {
              "height": 40,
              "margin": "4%",
              "resizeMode": "contain",
              "width": 40,
            }
          }
        />
        <Image
          source={
            Object {
              "testUri": "../../../App/Images/linkedinlogo.png",
            }
          }
          style={
            Object {
              "height": 40,
              "margin": "4%",
              "resizeMode": "contain",
              "width": 40,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "flexDirection": "row",
            "height": 82,
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#5D83AE",
              "fontSize": 26.5,
              "paddingLeft": "4%",
              "width": "50%",
            }
          }
        >
          Privacy Policy
        </Text>
        <Text
          style={
            Object {
              "color": "#5D83AE",
              "fontSize": 26.5,
              "paddingLeft": "4%",
              "width": "50%",
            }
          }
        >
          Fund Documents
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "flexDirection": "row",
            "height": 82,
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#5D83AE",
              "fontSize": 26.5,
              "paddingLeft": "4%",
              "width": "50%",
            }
          }
        >
          User Agreement
        </Text>
        <Text
          style={
            Object {
              "color": "#5D83AE",
              "fontSize": 26.5,
              "paddingLeft": "4%",
              "width": "50%",
            }
          }
        >
          Support
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#56565A",
            "height": 82,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
            }
          }
        >
          Copyright Victory Capital Management Inc. ©2020
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
